.github/workflows/main.yml

```diff
@@ -44,5 +47,39 @@ jobs:
       - name: Run Next.js build to generate the static files
         run: npm run build
 
-  # TODO: Deployment job
-  # TODO: Health check job
+        # Runs the deploy command
+        # Parses the output, and 
+        # retrieves the build log url & the deployed url,
+        # finally stores them into variables: NETLIFY_LOGS_URL & NETLIFY_LIVE_URL
+      - name: 🚀Deploy
+        id: deploy-netlify
+        run: |
+          COMMAND="netlify deploy --build --prod"
+          OUTPUT=$(sh -c "$COMMAND")
+
+          NETLIFY_LOGS_URL=$(echo "$OUTPUT" \
+          | grep -Eo '(http|https)://app.netlify.com/[a-zA-Z0-9./?=_-]*')
+          # Unique key: app.netlify.com
+          
+          NETLIFY_LIVE_URL=$(echo "$OUTPUT" \
+          | grep -Eo '(http|https)://[a-zA-Z0-9./?=_-]*' | grep -Ev "netlify.com|(--)")
+          # Unique key: doesn't contain -- and app.netlify.com
+
+          echo "::set-output name=netlify_logs_url::$NETLIFY_LOGS_URL"
+          echo "::set-output name=netlify_live_url::$NETLIFY_LIVE_URL"
+        env:
+          NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
+          NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+
+       # Takes the Logs url and live url from previous job
+       # and creates a comment on commit
+      - name: 👓 Netlify Preview URL
+        uses: peter-evans/commit-comment@v1
+        env:
+            OUTPUT: |
+              ### Successfully deployed to Netlify.
+
+              🔍 Inspect: ${{ steps.deploy-netlify.outputs.NETLIFY_LOGS_URL }}
+              ✅ Preview: ${{ steps.deploy-netlify.outputs.NETLIFY_LIVE_URL }}
+        with:
+          body: ${{ env.OUTPUT }}
```

Comment:
Let's keep this workflow simple? For more context, right now we push directly to `master` to get the project moving forward as fast as possible. But once the project is stable enough we plan to push to a `develop` branch as well. Imagine how often our notifications inbox will be spammed with "deployment notifications" on each commits pushed! :sweat_smile: 

I suggest refactor this workflow to only deploy to Netlify when a push event has occurred on the `master` branch. On the other hand, commenting on a PR with deployment messages is completely acceptable! So, the way you're written the `preview.yml` is good enough.

Reply:
Sorry, but I didn't really get what you're saying. The main workflow currently only deploys production build on push to the master branch. So, could you please elaborate ? Thanks

Reply:
TL;DR: Keep the deployment job simple, no need to add deployment comments to individual commits.